File: dotnet/src/FleetLink.Api/Models/Depot.cs
namespace FleetLink.Api.Models;

/// <summary>Depot — FSD §3.1. Where a vehicle is based and a work order is carried out.</summary>
public class Depot
{
    public Guid Id { get; set; }
    public string Code { get; set; } = default!;   // unique business key, e.g. DEP-LDN
    public string Name { get; set; } = default!;
    public string City { get; set; } = default!;

    // Navigation (FSD §4: Depot 1—* Vehicle, Depot 1—* Driver).
    public ICollection<Vehicle> Vehicles { get; set; } = new List<Vehicle>();
    public ICollection<Driver> Drivers { get; set; } = new List<Driver>();
}

File: javascript/src/models/depot.js
// Depot — FSD §3.1. A location a vehicle is based at and a work order is carried out at.
// Factory only: it names the shape. No persistence and no business rules here (those are 2.D).
export function makeDepot({ id, code, name, city }) {
  return { id, code, name, city };
}
